src/components/Tool/FixPhoneNumbers/FixPhoneNumbers.test.tsx

```diff
@@ -82,120 +82,142 @@ describe('FixPhoneNumbers-Home', () => {
-    const { getByText, getByTestId } = render(<Components data={[]} />);
-    await waitFor(() =>
-      expect(getByTestId('fixPhoneNumbers-null-state')).toBeInTheDocument(),
-    );
+    const { getByText, findByTestId } = render(<Components data={[]} />);
+
+    expect(
+      await findByTestId('fixPhoneNumbers-null-state'),
+    ).toBeInTheDocument();
+
     expect(
       getByText('No people with phone numbers need attention'),
     ).toBeInTheDocument();
   });
 
   it('should modify first number of first contact', async () => {
-    const { getByTestId } = render(<Components />);
-    await waitFor(() => {
-      expect(getByTestId('textfield-testid-0')).toBeInTheDocument();
-    });
-    const firstInput = getByTestId('textfield-testid-0') as HTMLInputElement;
+    const { getByTestId, findByTestId } = render(<Components />);
+
+    expect(await findByTestId('textfield-testid-id1')).toBeInTheDocument();
+    const firstInput = getByTestId('textfield-testid-id1') as HTMLInputElement;
 
-    expect(firstInput.value).toBe('+3533895895');
+    expect(firstInput.value).toBe('+353');
     userEvent.type(firstInput, '123');
-    expect(firstInput.value).toBe('+3533895895123');
+    await waitFor(() => {
+      expect(firstInput).toHaveValue('+353');
+    });
   });
 
   it('should hide contact from view', async () => {
-    const { getByTestId, getByText } = render(<Components />);
-    await waitFor(() => {
-      expect(getByText(`Simba Lion`)).toBeInTheDocument();
-    });
+    const { getByTestId, findByText } = render(<Components />);
+    expect(await findByText(`Simba Lion`)).toBeInTheDocument();
 
     userEvent.click(getByTestId('confirmButton-testid'));
+
     await waitFor(() => {
-      expect(mockEnqueue).toHaveBeenCalledWith('Phone numbers updated!', {
-        variant: 'success',
-      });
+      expect(mockEnqueue).toHaveBeenCalledWith(
+        'Successfully updated phone numbers for Test Contact',
+        {
+          variant: 'success',
+        },
+      );
     });
   });
   it('should bulk confirm all phone numbers', async () => {
-    const { getByTestId, queryByTestId, getByText } = render(<Components />);
+    const { getByTestId, queryByTestId, getByText, findByTestId, getByRole } =
+      render(<Components />);
     await waitFor(() => {
       expect(queryByTestId('loading')).not.toBeInTheDocument();
-      expect(getByTestId('starOutlineIcon-testid-1')).toBeInTheDocument();
     });
 
-    userEvent.click(getByTestId(`starOutlineIcon-testid-1`));
+    expect(
+      await findByTestId('starOutlineIcon-testid-id2'),
+    ).toBeInTheDocument();
+
+    userEvent.click(getByTestId(`starOutlineIcon-testid-id2`));
+
+    const primarySource = getByRole('combobox', {
+      name: 'Default Primary Source:',
+    });
+
+    userEvent.click(primarySource);
+    userEvent.click(getByRole('option', { name: 'MPDX' }));
+
+    expect(getByTestId('source-select')).toHaveValue('MPDX');
```

Comment:
```suggestion
    expect(primarySource).toHaveValue('MPDX');
```

Reply:
Honestly, I was also thinking that you could remove this line of the test. A more useful test could be to check that the confirm button as the text "Confirm 52 as MPDX". And I just realized that if "MPDX" is the initial value of that combobox, it would be better to select a different option so that the value actually changes. I feel like I'm stringing you along in asking for changes to the tests, but I promise that's not my intention.